### PR TITLE
Fix18806

### PR DESCRIPTION
**Pull request by bendk**
Bunch of fixes for #18806.  I'd love review, but please don't merge until Janet says that its fast enough on her machine.

### CODE REVIEW

File: tv/lib/metadata.py [hunk 1]
```diff
@@ -698,6 +713,38 @@ def _should_process_codegen_queue(self):
...
+                                      self._restart_after_http_errors, name)
+                self._waiting_from_http_errors = True
+
+    def _process_metadata_fetch_queue(self):
+        while self._metadata_fetch_queue:
+            path, metadata_fetcher = self._metadata_fetch_queue.pop()
+            try:
+                metadata = metadata_fetcher()
+            except KeyError:
+                logging.warn("_process_metadata_fetch_queue: metadata_fetcher "
```
Comment (paulswartz) [inline tv/lib/metadata.py:739]: Is this log item going to be useful? Some things I can think to include:
- item
- fetcher function
- traceback

Also, what about other exceptions?

Comment (bendk) [inline tv/lib/metadata.py:739]: Yeah, all of those things would be good.  By item, I'm going to use path.

I was thinking that other exceptions should be propagated upward, but now I don't think so anymore.  I'll change to catch StandardError.

**Comment by bendk** [comment]
Also, janet checked out the branch and it was running fast enough for her.  So this can be merged once it passes review.

File: tv/lib/metadata.py [hunk 2]
```diff
@@ -1471,23 +1527,25 @@ def _run_movie_data(self, path):
...
-        # we only send a subset of the metadata to echonest and some of the
-        # key names are different
-        echonest_metadata = {}
-        for key in ('title', 'artist', 'album', 'duration'):
-            try:
-                echonest_metadata[key] = metadata[key]
-            except KeyError:
-                pass
-        self.echonest_processor.add_path(path, echonest_metadata)
+        def metadata_fetcher():
```
Comment (paulswartz) [inline tv/lib/metadata.py:1532]: Do we still need this comment, since we're doing it later to avoid the slowdown?

**Comment by paulswartz** [comment]
Looks good!

**Pull request merged**